tecan_od_analyzer/tecan_od_analyzer.py

```diff
@@ -402,10 +403,28 @@ def compensation_lm(cor_df, df_gr) : #done
 	#Use the linear models to correct the volume loss by bioshaker
 	df_gr_comp = pd.DataFrame()
 	df_gr_comp_out = pd.DataFrame()
+	# prepare df for background correction
+	df_gr_background = pd.DataFrame()
 	for pos in range(len(lm_eq)) :
 		df_gr_comp = df_gr[df_gr["bioshaker"] ==  unique_bioshaker[pos]]
-		df_gr_comp["Correlation"] = lm_eq[pos][0]*df_gr_comp["time_hours"]+lm_eq[pos][1]
-		df_gr_comp["Measurement"] = df_gr_comp["Measurement"]*df_gr_comp["Correlation"]
+		
+		# The following two loops are there for background correction
+		# get OD600 values of volume correction samples to use for background correction
+		df_gr_background = df_vl600[df_vl600["bioshaker"] ==  unique_bioshaker[pos]]
+		sample_IDs = list(df_gr_comp['Sample_ID'].unique())
+		for sample in sample_IDs:
+			df_gr_comp_sample_limited = df_gr_comp[df_gr_comp['Sample_ID'] == sample]
+			times = list(df_gr_comp['time_hours'][df_gr_comp['Sample_ID'] == sample].unique())
+			for time in times:
+				df_gr_comp_sample_time_limited = df_gr_comp_sample_limited[df_gr_comp_sample_limited['time_hours'] == time]
+				df_gr_comp_index = df_gr_comp_sample_time_limited.index[0]
+				df_gr_background_time_limited = df_gr_background[df_gr_background['time_hours'] == time]
+				background_mean = np.mean(df_gr_background_time_limited["Measurement"])
+				df_gr_comp["Measurement"].loc[df_gr_comp_index] = df_gr_comp["Measurement"].loc[df_gr_comp_index] - background_mean
+
+		df_gr_comp["Correlation"] = lm_eq[pos][0]*df_gr_comp["time_hours"]+lm_eq[pos][1] # Should this just be 1? Or the normalized initial value
+		#df_gr_comp["Measurement"] = df_gr_comp["Measurement"]*df_gr_comp["Correlation"] # I guess this should be divided rather than multiplied
```

Comment:
I think the equation needs to be inversed, correct?  So it would be division?

Reply:
I think in that case it needs to be multiplied to get the correlation which is used to divide the values in the next step. Division would also produce an error (division by zero) since we're using the time values and have measurements from time point zero.
If you are referring to the calculation that produces the df_gr_comp["Measurement"] value, I already adjusted everything to the way I describe it in the comments above, guess it just got lost somewhere in the other commits